### PR TITLE
Fix build without setns

### PR DESCRIPTION
**Pull request by loveshack**
You better check this is right, but it builds on el6.

### CODE REVIEW

**Comment by gmkurtzer** [comment]
Heya Dave, this looks good. About my only question I have so far is with the debootstrap min base variant. Inconsistently during my testing with different versions I noticed that debootstrap would sometimes omit apt so I added that in the include list. Have you seen similar?

**Comment by loveshack** [comment]
git(hub) is confusing me no end.  I didn't mean to push the debootstrap change as it had developed a conflict, and I've just resolved that while trying to make another request for docker import that looks as if it's now mixed in.

I'm surprised if apt isn't implied. With which version of Debian? It's definitely "essential" in jessie, and requires 'Yes, do as I say!' to remove, but maybe that depends on debootstrap -- I haven't checked how it works.  There could be a case for not including it if you didn't want to pull more packages, of course, and it's a pity there isn't something like yum's --installroot as far as I know.

**Comment by loveshack** [comment]
I was going to comment on the import change.  Grr.

It implements the suggestion to have import support different inputs, but has some rough edges, like wanting format-specific args.  I don't know whether it's the best approach but don't have strong feelings.
I'm still puzzled by the required size of the singularity container relative to the tar size or filesystem usage and will try to figure that out.

**Comment by gmkurtzer** [comment]
If I recall correctly, it was specific to a particular debootstrap version that would omit apt if I specified minbase. I don't think it hurts to specify it as an include just to be sure. 

Yeah, some of the fork/pull/remote features of github I'm still struggling with as well. My background is almost all with single master repos like CVS and SVN. 

As far as the size goes, there will just need to be a fudge factor. Also considering the file system overhead plus you don't want it to be exact, here should always be a bit of wiggle room after the import is done. 

Do you want me to accept this pull request now, or wait for you to adjust a few more things?

Oh one last bit. I added a documentation page on contributing code and as it pertains to asserting copyright. I agree with your choice here but the only thing my legals told me to always make sure is that the projects copyright is also included in a contributed file.

Thanks!

**Comment by loveshack** [comment]
I've pushed a version with the header.
I won't do any more on the import soon.  Feel free to grab it if you think it's OK-ish.

File: libexec/docker-import.sh [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+# scratch file, but perhaps the file would be better.
+# Note that the Docker semantics for the entrypoint and cmd aren't too
+# clear <https://docs.docker.com/engine/reference/builder/>.
+
+if ! cmd=$(docker inspect --format='{{json .Config.Cmd}}' "$dock"); then
+    message ERROR "docker inspect failed\n"
+    exit 1
+fi
+
+entry='["/bin/sh","-c"]'
```
Comment (gmkurtzer) [inline libexec/docker-import.sh]: Why do you statically define the entry point and command here?

Comment (loveshack) [inline libexec/docker-import.sh]: [Sorry this didn't get sent.  Yesterday must have been a while ago!]

You wrote: 

> > +entry='["/bin/sh","-c"]'
> 
> Why do you statically define the entry point and command here?

Sorry, I checked in a version with that for testing.  I pushed a change
yesterday, but didn't get round to the pull request -- though I don't
understand how to refer to specific changes.

**Pull request merged**